File: An2/Sem1/MAP/MAP-C#/MAP-Lab9/MAP-Lab9/Model/PhDStudent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MAP_Lab9.Model
{
    [Serializable()]
    class PhDStudent : Student
    {
        public String supervisor;
        public String thesis;
        public int grade2;

        public PhDStudent(int id, String name, int grade, String supervisor, String thesis, int grade2)
            : base(id, name, grade)
        {
            this.supervisor = supervisor;
            this.thesis = thesis;
            this.grade2 = grade2;
        }

        public override float average()
        {
            return (this.getGrade() + this.grade2) / 2;
        }


        public override String ToString()
        {
            return this.getID() + " " + this.getName() + " " + this.getGrade() + " " + this.supervisor + " " + this.thesis + " " + this.grade2;
        }
    }
}

File: An2/Sem1/MAP/MAP-Java/MAP-Lab10/src/Model/PhDStudent.java
package Model;

/**
 * Created with IntelliJ IDEA.
 * User: mihai
 * Date: 12/4/13
 * Time: 11:46 PM
 * To change this template use File | Settings | File Templates.
 */
public class PhDStudent extends Student {

    public String supervisor;
    public String thesis;
    public int grade2;

    public PhDStudent(){}

    public PhDStudent(int id, String name, int grade, String supervisor, String thesis, int grade2){
        super(id, name, grade);
        this.supervisor = supervisor;
        this.thesis = thesis;
        this.grade2 = grade2;
    }

    @Override
    public float average(){
        return (this.getGrade() + grade2) / 2;
    }

    @Override
    public boolean isGreaterThan(Student student){
        return (this.average() > student.average());
    }

    @Override
    public String toString(){
        return this.getID()+" "+this.getName()+" "+this.getGrade()+" "+supervisor+" "+thesis+" "+grade2;
    }

}
